set.mm

```diff
@@ -660473,6 +660392,215 @@ D Fn ( I ... ( M - 1 ) ) /\
+    $( Lemma for ~ rhmmpl .  A subproof of ~ psrmulcllem .  (Contributed by SN,
+       8-Feb-2025.) $)
+    rhmmpllem2 $p |- ( ( ph /\ k e. D ) ->
+      ( R gsum ( x e. { y e. D | y oR <_ k } |->
+        ( ( X ` x ) ( .r ` R ) ( Y ` ( k oF - x ) ) ) ) ) e. ( Base ` R ) ) $=
+      ( cv wcel wa wbr cfv eqid cle cofr crab cbs cmin cof co cmpt cfn c0g ccmn
+      cmulr ringcmnd adantr psrbaglefi adantl ad2antrr wf breq1 elrab ffvelcdmd
+      crg biimpi simpld cn0 simplr psrbagf syl simprd psrbagcon syl3anc ringcld
+      fmpttd rhmmpllem1 gsumcl ) AGOZDPZQZCOZVPUAUBZRZCDUCZEUDSZBWBBOZISZVPWDUE
+      UFUGZJSZEULSZUGZUHEUIEUJSZWCTZWJTAEUKPVQAELUMUNVQWBUIPACDFVPHKUOUPVRBWBWI
+      WCVRWDWBPZQZWCEWHWEWGWKWHTAEVBPVQWLLUQWMDWCWDIADWCIURVQWLMUQWMWDDPZWDVPVT
+      RZWLWNWOQZVRWLWPWAWOCWDDVSWDVPVTUSUTVCUPZVDZVAWMDWCWFJADWCJURVQWLNUQWMWFD
+      PZWFVPVTRZWMVQHVEWDURZWOWSWTQAVQWLVFWMWNXAWRDFWDHKVGVHWMWNWOWQVIDFVPWDHKV
+      JVKVDVAVLVMABCDEFGHIJKLMNVNVO $.
+  $}
+
+  ${
+    $d I f $.
+    mhmcoaddmpl.p $e |- P = ( I mPoly R ) $.
+    mhmcoaddmpl.q $e |- Q = ( I mPoly S ) $.
+    mhmcoaddmpl.b $e |- B = ( Base ` P ) $.
+    mhmcoaddmpl.c $e |- C = ( Base ` Q ) $.
+    mhmcoaddmpl.1 $e |- .+ = ( +g ` P ) $.
+    mhmcoaddmpl.2 $e |- .+b = ( +g ` Q ) $.
+    mhmcoaddmpl.i $e |- ( ph -> I e. V ) $.
+    mhmcoaddmpl.h $e |- ( ph -> H e. ( R MndHom S ) ) $.
+    mhmcoaddmpl.f $e |- ( ph -> F e. B ) $.
+    mhmcoaddmpl.g $e |- ( ph -> G e. B ) $.
+    $( The ring homomorphism in ~ rhmmpl preserves addition.  (Contributed by
+       SN, 8-Feb-2025.) $)
+    mhmcoaddmpl $p |- ( ph -> ( H o. ( F .+ G ) ) =
+                              ( ( H o. F ) .+b ( H o. G ) ) ) $=
+      ( vf cplusg cfv cof co ccom cmhm wcel cbs cv ccnv cima cfn cmap crab wceq
+      cn cn0 cvv fvexd ovex rabex a1i eqid mplelf elmapdd mhmvlin mpladd coeq2d
+      syl3anc mhmcompl 3eqtr4d ) ALJKHUFUGZUHUIZUJZLJUJZLKUJZIUFUGZUHUIZLJKEUIZ
+      UJVTWAFUIALHIUKUIULJHUMUGZUEUNUOVAUPUQULZUEVBMURUIZUSZURUIZULKWIULVSWCUTU
+      BAWEWHJVCVCAHUMVDZWHVCULAWFUEWGVBMURVEVFVGZABWHDHUEMWEJOWEVHZQWHVHZUCVIVJ
+      AWEWHKVCVCWJWKABWHDHUEMWEKOWLQWMUDVIVJWEVQWBLWHHIJKWLVQVHZWBVHZVKVNAWDVRL
+      ABDVQEHMJKOQWNSUCUDVLVMACGWBFIMVTWAPRWOTABCDGHIJLMNOPQRUAUBUCVOABCDGHIKLM
+      NOPQRUAUBUDVOVLVP $.
+  $}
+
+  ${
+    $d ph d k $.  $d R d k $.  $d S d k $.  $d F d k $.  $d G d k $.
+    $d H d k $.  $d I d e f k $.
+    rhmcomulmpl.p $e |- P = ( I mPoly R ) $.
+    rhmcomulmpl.q $e |- Q = ( I mPoly S ) $.
+    rhmcomulmpl.b $e |- B = ( Base ` P ) $.
+    rhmcomulmpl.c $e |- C = ( Base ` Q ) $.
+    rhmcomulmpl.1 $e |- .x. = ( .r ` P ) $.
+    rhmcomulmpl.2 $e |- .xb = ( .r ` Q ) $.
+    rhmcomulmpl.i $e |- ( ph -> I e. V ) $.
+    rhmcomulmpl.h $e |- ( ph -> H e. ( R RingHom S ) ) $.
+    rhmcomulmpl.f $e |- ( ph -> F e. B ) $.
+    rhmcomulmpl.g $e |- ( ph -> G e. B ) $.
+    $( The ring homomorphism in ~ rhmmpl preserves multiplication.
+       (Contributed by SN, 8-Feb-2025.) $)
```

Comment:
The comment is a bit confusing because all ring homomorphisms preserve multiplication.
But at this point we don't know that that function is a ring homomorphism.
Maybe "Lemma for ~ rhmmpl : multiplication is preserved" ?
(same for the addition)

Reply:
Well, multiplication is preserved is just another way to say preserves multiplication. And it's not a lemma since it's used in selvcllem4 (and selvmul when it gets added).

If saying something is true for one case kind of implies somehow as an informal language construct that it is not true for other cases, then maybe adding "Show that" to the start would work, to neutralize the misconception by "reframing" the statement away from such an interpretation?

Reply:
Yes, maybe it would help.
Or maybe use "the function in ~ rhmmpl", or "the mapping in ~ rhmmpl", instead of "the ring homomorphism in ~ rhmmpl".
Anyway not a major issue!